core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -940,9 +940,16 @@ public Weighting createWeighting( WeightingMap weightingMap, FlagEncoder encoder
                 return new PriorityWeighting(encoder, weightingMap);
             else
                 return new FastestWeighting(encoder, weightingMap);
+        } else if ("curvature".equalsIgnoreCase(weighting) || weighting.isEmpty())
```

Comment:
weighting.isEmpty should be removed here (and is already handled one if before) ... I make this in the squashed commit ...